client/my-sites/site-settings/date-time-format/date-time-format-options.jsx

```diff
@@ -0,0 +1,338 @@
+import { includes, startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import ExternalLink from 'components/external-link';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormInput from 'components/forms/form-text-input';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import FormSelect from 'components/forms/form-select';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import SectionHeader from 'components/section-header';
+import { phpToMomentDatetimeFormat } from 'lib/formatting';
+import wrapSettingsForm from '../wrap-settings-form';
+
+/**
+ * Module constants
+ */
+const defaultDateFormats = [ 'F j, Y', 'Y-m-d', 'm/d/Y', 'd/m/Y' ];
+const defaultTimeFormats = [ 'g:i a', 'g:i A', 'H:i' ];
+
+export class DateTimeFormatOptions extends Component {
+	state = {
+		customDateFormat: false,
+		customTimeFormat: false,
+		isLoadingSettings: true,
+	};
+
+	isDefaultFormat = ( defaultFormats, format ) => ! includes( defaultFormats, format );
+
+	handleRadio = event => {
+		const { name: field, value: format } = event.currentTarget;
+		this.props.handleRadio( event );
+		if ( 'date_format' === field ) {
+			this.setState( {
+				customDateFormat: this.isDefaultFormat( defaultDateFormats, format ),
+			} );
+		} else if ( 'time_format' === field ) {
+			this.setState( {
+				customTimeFormat: this.isDefaultFormat( defaultTimeFormats, format ),
+			} );
+		}
+	}
+
+	handleInput = field => event => {
+		this.props.onChangeField( field )( event );
+		if ( 'date_format' === field ) {
+			this.setState( { customDateFormat: true } );
+		} else if ( 'time_format' === field ) {
+			this.setState( { customTimeFormat: true } );
+		}
+	}
+
+	handleCustomRadio = event => {
+		const { name: field } = event.currentTarget;
+		if ( 'date_format' === field ) {
+			this.setState( { customDateFormat: true } );
+		} else if ( 'time_format' === field ) {
+			this.setState( { customTimeFormat: true } );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const {
+			fields: {
+				date_format: dateFormat,
+				time_format: timeFormat,
+			},
+		} = newProps;
+
+		if ( ! this.state.isLoadingSettings || '' === dateFormat || '' === timeFormat ) {
+			return;
+		}
+
+		this.setState( {
+			customDateFormat: this.isDefaultFormat( defaultDateFormats, dateFormat ),
+			customTimeFormat: this.isDefaultFormat( defaultTimeFormats, timeFormat ),
+			isLoadingSettings: false,
+		} );
+	}
+
+	dateFormatOption() {
+		const {
+			fields: {
+				date_format: dateFormat,
+				timezone_string: timezoneString,
+			},
+			isRequestingSettings,
+			moment,
+			translate,
+		} = this.props;
+		const { customDateFormat: isCustomFormat } = this.state;
+
+		const today = startsWith( timezoneString, 'UTC' )
+			? moment().utcOffset( timezoneString.substring( 3 ) * 60 )
```

Comment:
a comment showing anticipated input data here would be incredibly helpful. that is to say, it's not at all obvious _why_ we're taking the substring and multiplying by 60, but if we had in a comment one possible time string then we could jump over with our eyes and see, "oh yeah, we're just skipping the _ and doing _ to it"